classes/Optimization/Process/AbstractProcess.php

```diff
@@ -1447,7 +1454,7 @@ protected function delete_nextgen_file( $file_path ) {
 		$next_gen_file = new File( $file_path );
 
 		// Delete next-gen images.
-		foreach ( $this->extensions as $extension ) {
+		foreach ( imagify_nextgen_images_formats() as $extension ) {
```

Comment:
This needs to stop being changed, on a previous PR we used the `$extensions` array because we want to remove all next-gen files when deleting the file, whatever is enabled at that time.

Why is it changed back to the function that will only return formats that are currently enabled?

Reply:
According to @Mai-Saad , her first feedback, she said we were deleting the webp when we activate AVIF, and AVIF would be generated afterwards. Therefore, we would need to change the behaviour to not delete the webp when we generate the avif. 

When looking at why it gets deleted, when an optimization is started, we get to a point where https://github.com/wp-media/imagify-plugin/blob/a3e791d057ed0acff4f2af03a943c322a3d3465c/classes/Optimization/Process/AbstractProcess.php#L874 this function is called. 
and from this function we get : 
https://github.com/wp-media/imagify-plugin/blob/a3e791d057ed0acff4f2af03a943c322a3d3465c/classes/Optimization/Process/AbstractProcess.php#L1449


If we use `$this->extensions` we deletes every format that are included in this array:
https://github.com/wp-media/imagify-plugin/blob/a3e791d057ed0acff4f2af03a943c322a3d3465c/classes/Optimization/Process/AbstractProcess.php#L113-L116

Reply:
In my point of view, `webp` files should be deleted when we are enabling `avif` and the opposite as well. Otherwise we would use disk space for nothing with one format. But that's another story. 

I did not think that this function was used to delete the file..

Reply:
I was thinking about adding a param to the function yea, let me take a look at the code and make a commit :)